File: Atcoder_#319/javascript/b_measure.js
function main(input) {
  const n = parseInt(input.trim(), 10);
  const result = [];

  for (let i = 0; i <= n; i++) {
    let found = false;
    
    for (let j = 1; j <= 9; j++) {
      if (n % j === 0 && i % (n / j) === 0) {
        result.push(j.toString());
        found = true;
        break;
      }
    }
    
    if (!found) {
      result.push('-');
    }
  }

  console.log(result.join(''));
}

main(require('fs').readFileSync('/dev/stdin', 'utf8'));

File: Atcoder_#319/ruby/b_measure.rb
n = gets.to_i

result = []

(0..n).each do |i|
  found = false
  (1..9).each do |j|
    if n % j == 0 && i % (n / j) == 0
      result << j.to_s
      found = true
      break
    end
  end
  result << "-" unless found
end

puts result.join
